### PR TITLE
Support full Gradle build.

### PR DESCRIPTION
**Pull request by felixvf**
Up to now, building libsodium-jni for different architectures required executing many different scripts. This pull request alleviates this by providing a complete "build.gradle" file. This allows downstream consumers of libsodium-jni who also use gradle (e.g. Android apps) to use libsodium-jni seamlessly.

The "build.gradle" builds libsodium-jni for all Android architectures as well as for the architecture of the host OS. The latter facilitates testing. Running "gradle test" (testing on the host OS) is supported, as is "gradle connectedCheck" (testing on the emulator). This has also revealed some incompatibility bugs, which have been fixed.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -20,9 +24,9 @@ android {
     }
 
     // To let the libs be included in other projects
-    sourceSets {
```
Comment (joshjdevl) [inline build.gradle]: Are the shared libraries included or imported some other way now?

**Comment by joshjdevl** [comment]
Thanks for putting this together!

File: build.gradle (hunk 1 above)
Comment (felixvf) [inline build.gradle]: The shared libraries for Android (armeabi, arm64-v8a, x86, ...) are already automatically included in the generated .aar files.
However, the shared libraries for the host OS are not included in the .aar files or in any other .jar file. Should this change?

Comment (joshjdevl) [inline build.gradle]: > The shared libraries for Android (armeabi, arm64-v8a, x86, ...) are already automatically included in the generated .aar files.

Great!

> 
However, the shared libraries for the host OS are not included in the .aar files or in any other .jar file. Should this change?

If it's possible to do include the host OS, sure please go ahead? That would help make things potentially easier for the users of the library.

Comment (felixvf) [inline build.gradle]: Unfortunately, it seems that, unlike for Android platforms, there is no common standard on

1. how to cross compile JNI libraries
2. how to package cross compiled JNI libraries (and how to name the platforms such that compatible platforms have the same name and incompatible platforms have a different name)
3. how to load cross compiled JNI libraries
4. how to test cross compiled JNI libraries for all supported target platforms

on these platforms:

1. Linux x86,
2. Linux x86-64,
3. Linux armv6,
4. Linux armv7,
5. Linux armv8-a,
6. Windows x86,
7. Windows x86-64, 
8. MacOS universal binary

At least I’m not aware of such a standard. It also seems that creating and implementing such a standard is substantial work, as I can imagine that

1. on certain platforms, libraries need to be unpacked (from .jar files) before they can be loaded
2. the probability of cross compiler toolchains being available to the average developer for all target platforms is low, leading to build failures if we attempt to implement this, unless we first download and compile the cross compiler toolchains for all target platforms...

That’s why I’d consider this a separate issue.

**Pull request merged**